src/components/Imports/Providers/nextcloud/provider.js

```diff
@@ -0,0 +1,449 @@
+  },
+
+  async listRemote(client, accountId, path = '/') {
+    const clean = this._joinRemotePath(path)
+    const encodedPath = this._encodeRemotePath(clean)
+    const url = `/remote/nextcloud/${encodeURIComponent(
+      accountId
+    )}${encodedPath}`
+
+    const res = await client.stackClient.fetch('GET', url)
+
+    if (res.status === 401) {
+      const err = new Error('Unauthorized to Nextcloud')
+      err.status = 401
+      err.path = clean
+      throw err
+    }
+
+    if (res.status === 404) {
+      const txt = await res.text().catch(() => '')
+      const err = new Error(
+        txt ||
+          `${this._extractName(clean)} - 404, ${this._build404Reason(clean)}`
+      )
+      err.status = 404
+      err.path = clean
+      throw err
+    }
+
+    if (res.status !== 200) {
+      const txt = await res.text().catch(() => '')
+      const err = new Error(txt || `Remote list failed: HTTP ${res.status}`)
+      err.status = res.status
+      err.path = clean
+      throw err
+    }
+
+    const json = await res.json()
+    return json?.data || []
+  },
+
+  async findChildDirByName(client, parentId, name) {
+    const res = await client.stackClient.fetchJSON(
+      'POST',
+      '/data/io.cozy.files/_find',
+      { selector: { dir_id: parentId, name, type: 'directory' }, limit: 1 }
+    )
+    return res?.docs?.[0] || null
+  },
+
+  async createChildDir(client, parentId, name) {
+    const qs = new URLSearchParams({ Type: 'directory', Name: name })
+    const url = `/files/${encodeURIComponent(parentId)}?${qs.toString()}`
+    const res = await client.stackClient.fetch('POST', url, null, {
+      headers: { Accept: 'application/vnd.api+json' }
+    })
+
+    if (res.status >= 200 && res.status < 300) {
+      const json = await res.json().catch(() => ({}))
+      return json?.data || json
+    }
+
+    if (res.status === 409) {
+      const existing = await this.findChildDirByName(client, parentId, name)
+      if (existing) return existing
+    }
+
+    const txt = await res.text().catch(() => '')
+    throw new Error(txt || `Create directory failed: HTTP ${res.status}`)
+  },
+
+  async ensureChildDir(client, parentId, name) {
+    const existing = await this.findChildDirByName(client, parentId, name)
+    if (existing) return existing
+    return this.createChildDir(client, parentId, name)
+  },
+
+  async ensureImportsDestination(
+    client,
+    providerLabel = 'Nextcloud',
+    login = ''
+  ) {
+    const ROOT = 'io.cozy.files.root-dir'
+    const importsDir = await this.ensureChildDir(client, ROOT, 'Imports')
+    const providerDir = await this.ensureChildDir(
+      client,
+      importsDir._id || importsDir.id,
+      providerLabel
+    )
+    const safeLogin = String(login || 'unknown').replace(/[/\\]/g, '_')
+    const finalDir = await this.ensureChildDir(
+      client,
+      providerDir._id || providerDir.id,
+      safeLogin
+    )
+    return finalDir._id || finalDir.id
+  },
+
+  async downstreamFile(
```

Comment:
why not using https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/NextcloudFilesCollection.js#L138 ?